wallet/src/main/scala/org/bitcoins/wallet/api/CoinSelector.scala

```diff
@@ -0,0 +1,69 @@
+package org.bitcoins.wallet.api
+
+import org.bitcoins.core.currency.{CurrencyUnit, CurrencyUnits}
+import org.bitcoins.core.protocol.transaction.TransactionOutput
+import org.bitcoins.core.wallet.fee.FeeUnit
+import org.bitcoins.wallet.models.UTXOSpendingInfoDb
+
+import scala.annotation.tailrec
+
+/** Implements algorithms for selecting from a UTXO set to spend to an output set at a given fee rate. */
+trait CoinSelector {
```

Comment:
Can we come with a interface that works for _any_ coin selection scheme? So we can plug and play later on?

Reply:
I mean we could have an abstract `def select(walletUtxos: Vector[UTXOSpendingInfoDb], outputs: Vector[TransactionOutput], feeRate: FeeUnit): Vector[UTXOSpendingInfoDb]` as the only member of `CoinSelector` and then implement a bunch of different options as objects but I don't see any reason not just put them in one place so that its nice in an IDE to just type `CoinSelector.` and see what comes up, there's no real benefit to making an interface for this one method. I don't expect people to be passing in lambdas where coin selection algorithms are required (in fact that would be shady in my eyes)